test/command.cjs

```diff
@@ -1547,6 +1547,30 @@ describe('Command', () => {
             return done();
           });
       });
+
+      // addresses https://github.com/yargs/yargs/issues/1966
+      it('should not be applied multiple times', done => {
+        yargs('rule add foo bar baz')
+          .command('rule', 'rule desc', yargs =>
+            yargs.command(
+              'add <name> <abc...>',
+              'add desc',
+              yargs =>
+                yargs.positional('abc', {
+                  type: 'string',
+                  coerce: arg => arg.join(' '),
```

Comment:
I think it would be worth extending this function so that it throws if `coerce` is called more than once.

```js
let executionCount = 0
...
coerce: arg => {
  if (executionCount > 0) throw Error('coerce applied multiple times');
  executionCount++;
  return arg.join(' ');
};
```

I would then confirm that the test fails prior to your fix.